internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -290,5 +285,39 @@ func createWindowsContainerDocument(ctx context.Context, coi *createOptionsInter
 	}
 	v1.MappedPipes = mpsv1
 	v2Container.MappedPipes = mpsv2
+
+	// add assigned devices to the container definition
+	parseAssignedDevices(coi, v2Container)
 	return v1, v2Container, nil
 }
+
+func specHasAssignedDevices(coi *createOptionsInternal) bool {
+	if coi.Spec.Windows != nil && coi.Spec.Windows.Devices != nil &&
+		len(coi.Spec.Windows.Devices) > 0 {
+		return true
+	}
+	return false
+}
+
+// parseAssignedDevices parses assigned devices for the container definition
+// this is currently supported for v2 argon only
+func parseAssignedDevices(coi *createOptionsInternal, v2 *hcsschema.Container) {
+	if !specHasAssignedDevices(coi) || !coi.isV2Argon() {
+		return
+	}
+
+	v2AssignedDevices := []hcsschema.Device{}
+	for _, d := range coi.Spec.Windows.Devices {
```

Comment:
The documented `IDType` in OCI is `class` (per [here](https://github.com/opencontainers/runtime-spec/blob/master/config-windows.md#devices)). Is this the same thing as `vpciClassGUID`? Can we keep the name the same, so that the documented behavior in OCI continues to work?

Reply:
That is the same thing, yes. I'll update.